noxfile.py

```diff
@@ -15,25 +15,12 @@
 nox.needs_version = ">=2024.3.2"
 nox.options.default_venv_backend = "uv|virtualenv"
 
-PYTHON_ALL_VERSIONS = ["3.10", "3.11"]
+PYTHON_ALL_VERSIONS = ["3.9", "3.10", "3.11", "3.12"]
 
 BUILD_REQUIREMENTS = [
     "setuptools>=66.1",
-    "setuptools_scm>=7",
+    "setuptools_scm>=8.1",
     "wheel>=0.40",
```

Comment:
```suggestion
    "wheel>=0.40",
    "cython>=3.0",
```
This could actually fix the macOS issues.